provider/marathon/marathon.go

```diff
@@ -274,8 +357,24 @@ func (p *Provider) getLabel(application marathon.Application, label string) (str
 	return "", false
 }
 
-func (p *Provider) getPort(task marathon.Task, application marathon.Application) string {
-	port, err := processPorts(application, task)
+// Extract port from labels for a given service and a given marathon application
+func (p *Provider) getPort(task marathon.Task, application marathon.Application, serviceName string) string {
+	if value, ok := getServiceProperty(application, serviceName, "port"); ok {
```

Comment:
The extended implementation of this method is fairly hard to reason about. I also think it's not on par with what we had before: For instance, we do not do the same level of sanity-checking on a found service port as we do on an application port.

Can we refactor both `getPort` and `processPorts` in the following regards:

- Have `getPort` look for a service port. If one exists, have it look for a service port index and pass both to `processPorts`. If no service port is found, have it look for an application port and port index and pass those to `processPorts`.
- Have `processPorts` parse the given port and port index (i.e., remove the label extraction part from it so that it becomes reusable).